chatkit-core/src/main/kotlin/com/pusher/chatkit/CurrentUser.kt

```diff
@@ -37,47 +40,114 @@ class CurrentUser(
+     * The [position] is the id of the last message that the current user has read.
+     * This will consequently notify any other connected users who have a [RoomListeners]
+     * set for [onNewReadCursor] to let them know the current user has read up to [position].
+     */
     fun setReadCursor(room: Room, position: Int) {
         Futures.schedule { syncCurrentUser.setReadCursor(room.id, position) }
     }
 
+    /**
+     * @see [setReadCursor]
+     */
     fun setReadCursor(roomId: String, position: Int) {
         Futures.schedule { syncCurrentUser.setReadCursor(roomId, position) }
     }
 
+    /**
+     * Gets the current users read cursor for the [roomId]. This method is synchronous and you will
+     * receive a [Result] with a [Cursor] if the process was successful, or an [Error] informing you
+     * of what went wrong. A common error is you must first be subscribed to the room to see read cursors.
+     */
     fun getReadCursor(roomId: String): Result<Cursor, Error> =
             syncCurrentUser.getReadCursor(roomId)
-
+    /**
+     * @see [getReadCursor]
+     */
     fun getReadCursor(room: Room): Result<Cursor, Error> =
             syncCurrentUser.getReadCursor(room.id)
 
+    /**
+     * Gets the current read cursor for the [userId] in [roomId]. This method is synchronous and you will
+     * receive a [Result] with a [Cursor] if the process was successful, or an [Error] informing you
+     * of what went wrong. A common error is you must first be subscribed to the room to see read cursors.
+     */
     fun getReadCursor(roomId: String, userId: String): Result<Cursor, Error> =
             syncCurrentUser.getReadCursor(roomId, userId)
 
+    /**
+     * @see [getReadCursor]
+     */
     fun getReadCursor(room: Room, user: User): Result<Cursor, Error> =
             syncCurrentUser.getReadCursor(room.id, user.id)
 
+    /**
+     * Make the users listed in [userIds] members of the room identified by [roomId]. This does not mean
+     * the user(s) will see new messages immediately, it means the room will be in their list of rooms
+     * and they will need to subscribe themselves to see any updates.
+     * Calling this method requires the current user to have the room:members:add role.
+     * The [callback] will be called with a [Result] when the
+     * operation is complete. If the operation was successful there is no value returned, or an
+     * [Error] informing you of what went wrong.
+     */
     fun addUsersToRoom(roomId: String, userIds: List<String>, callback: (Result<Unit, Error>) -> Unit) =
             makeCallback({ syncCurrentUser.addUsersToRoom(roomId, userIds) }, callback)
 
+    /**
+     * Removes the [userIds] from the [roomId]. The [callback] will be called with a [Result] when the
+     * operation is complete. If the operation was successful there is no value returned, or an
+     * [Error] informing you of what went wrong.
```

Comment:
This should get the same treatment as add users to room above.

Reply:
I've added the part about needing the `room:members:remove` permission - is that what you meant?

Reply:
That's useful. I guess I meant the whole shebang - what does it mean to remove a user form a room and what will they experience, mirroring the description for add users to room.

Reply:
Oh! I see.

I updated the removeUsers method to contain:
```
For any users currently online they will be notified in the [RemovedFromRoom] event and you will need to
     * handle how this is displayed in the UI to the user. The current users list of rooms will also be updated
     * to remove this room.
```
and similiarly updated the added to room to match
```
* For any users currently only they will be notified with a [AddedToRoom] event and you will need to handle
     * subscribing to the room to see any new messages sent to the room. The current users list of rooms
     * will also be updated to include this room.
```